libnavigation-util/src/main/java/com/mapbox/navigation/utils/thread/WorkThreadHandler.kt

```diff
@@ -54,7 +50,7 @@ class WorkThreadHandler(
         try {
             handlerThread.join()
         } catch (e: InterruptedException) {
-            MapboxLogger.e(TAG, e.localizedMessage, e)
+            e.printStackTrace()
```

Comment:
We cannot use `MapboxLogger` reference, we need to first load the `Logger` module as it could've been replaced by the developer's custom implementation. 

`WorkThreadHandler` should either do its own module discovery or accept a `Logger` instance via the constructor. We shouldn't rely on default implementations of modules anywhere in our code, we should always build against interfaces.
Since there's not much going on besides printing an exception in this class, I opted to drop this dependency entirely for now.

Reply:
I see why we separate logger to different module, but didn't get why we suggest user to replace it. `MapboxLogger` has `setObserver` if user would like to receive lib's logs

Reply:
If, for example, the developer wouldn't like to bring in the Timber dependency that the `MapboxLogger` is based on, they can replace the whole module and provide a completely custom implementation of the `Logger` interface. That's why we can't use `MapboxLogger` directly.